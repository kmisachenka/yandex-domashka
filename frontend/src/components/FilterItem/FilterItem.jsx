import React from 'react';
import { useToggle } from 'react-use';
import * as propTypes from 'prop-types';
import hexToRgba from 'hex-to-rgba';
import styles from './FilterItem.module.scss';

export default function FilterItem(props) {
  const { checked } = props;
  const [isChecked, toggleChecked] = useToggle(checked);
  const { color, handleClick } = props;
  return (
    <label className={styles.label}>
      <input
        className={styles.input}
        type="checkbox"
        checked={isChecked}
        onChange={() => {
          toggleChecked();
          handleClick(isChecked);
        }}
      />
      <span
        className={styles.checkbox}
        style={{ backgroundColor: hexToRgba(color) }}
      />
    </label>
  );
}

FilterItem.defaultProps = {
  color: '#E84747',
  checked: false,
};

FilterItem.propTypes = {
  handleClick: propTypes.func.isRequired,
  color: propTypes.string,
  checked: propTypes.bool,
};
