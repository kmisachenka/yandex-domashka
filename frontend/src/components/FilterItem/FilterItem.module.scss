.label {
  display: block;
  height: 48px;
  width: 48px;
}

.input {
  clip: rect(0 0 0 0);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;

  &:checked + .checkbox {
    background-image: url('./svg/checked.svg');
    box-shadow: 0 2px 2px rgba(0, 0, 0, 0.25);
    opacity: 1;

    transition: opacity 0.2s ease-in, box-shadow 0.1s ease-in;
  }

  &:focus:not(:active) + .checkbox {
    box-shadow:
      0 0 0 0.1em #4a90e2,
      0 0 0 0.2em #7ed321;
  }
}

.checkbox {
  background-position: 50% 50%;

  background-repeat: no-repeat;
  border-radius: 8px;
  height: 48px;

  opacity: 0.4;
  overflow: hidden;
  position: absolute;
  width: 48px;

  &:hover {
    opacity: 1;

    transition: opacity 0.1s ease-in;
  }
}
