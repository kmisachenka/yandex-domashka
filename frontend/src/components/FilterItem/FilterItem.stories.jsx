import React from 'react';
import { storiesOf } from '@storybook/react';
import FilterItem from './FilterItem';

const noop = () => {};

storiesOf('Фильтр: элемент', module)
  .add('с дефолтным цветом', () => (
    <FilterItem handleClick={noop} />
  ))
  .add('с кастомным цветом', () => (
    <FilterItem color="#219653" handleClick={noop} />
  ))
  .add('checked', () => (
    <FilterItem checked handleClick={noop} />
  ));
