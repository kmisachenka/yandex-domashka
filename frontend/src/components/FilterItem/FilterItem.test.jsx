import React from 'react';
import { shallow } from 'enzyme';

import FilterItem from './FilterItem';

describe('FilterItem', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<FilterItem />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should be unchecked by default', () => {
    const wrapper = shallow(<FilterItem />);
    expect(wrapper).not.toBeChecked();
  });
  it('should handle onChange event', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<FilterItem onClick={onClick} />);
    wrapper.find('input').simulate('change');
    expect(onClick).toHaveBeenCalled();
  });

  it('should set checked if passed', () => {
    const wrapper = shallow(<FilterItem checked />);
    expect(wrapper.find('input')).toBeChecked();
  });
});
