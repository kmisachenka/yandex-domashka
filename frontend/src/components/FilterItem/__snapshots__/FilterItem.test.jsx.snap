// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterItem should match snapshot 1`] = `
<label
  className="label"
>
  <input
    checked={false}
    className="input"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="checkbox"
    style={
      Object {
        "backgroundColor": "rgba(232, 71, 71, 1)",
      }
    }
  />
</label>
`;
